internal/providers/vmware/vmware_amd64.go

```diff
@@ -97,3 +99,42 @@ func fetchRawConfig(f *resource.Fetcher) (config, error) {
 		encoding: encoding,
 	}, nil
 }
+
+func DelConfig(f *resource.Fetcher) error {
+	info := rpcvmx.NewConfig()
+
+	orig, err := info.String(GUESTINFO_USERDATA, "")
+	if err != nil {
+		return fmt.Errorf("getting config property: %w", err)
+	}
+	if orig != "" {
+		f.Logger.Info("deleting config from guestinfo properties")
+		// we can't delete properties or set them to the empty
+		// string, so set encoding to "base64" and data to encoded "{}"
+		if err := info.SetString(GUESTINFO_USERDATA, "e30K"); err != nil {
+			return fmt.Errorf("replacing config: %w", err)
+		}
+		if err := info.SetString(GUESTINFO_USERDATA_ENCODING, "base64"); err != nil {
+			return fmt.Errorf("replacing config encoding: %w", err)
+		}
+	}
+
+	ovfEnv, err := info.String(GUESTINFO_OVF, "")
+	if err != nil {
+		// unlike FetchConfig, don't ignore errors, since that could
+		// have security implications
+		return fmt.Errorf("reading OVF environment: %w", err)
+	}
+	if ovfEnv != "" {
+		f.Logger.Info("deleting config from OVF environment")
+		pruned, err := DeleteOvfProperties([]byte(ovfEnv), []string{OVF_USERDATA, OVF_USERDATA_ENCODING})
```

Comment:
And here too, the function could return a flag to say if it actually deleted anything so we can avoid the `SetString`.